interface/patient_tracker/patient_tracker.php

```diff
@@ -43,16 +43,36 @@
 
 #define variables, future enhancement allow changing the to_date and from_date 
 #to allow picking a date to review
- 
+
+$provider = !is_null($_POST['form_provider']) ? $_POST['form_provider'] : null;
+$facility  = !is_null($_POST['form_facility']) ? $_POST['form_facility'] : null;
+$form_apptstatus = !is_null($_POST['form_apptstatus']) ? $_POST['form_apptstatus'] : null;
+$form_apptcat=null;
+if(isset($_POST['form_apptcat']))
+{
+    if($form_apptcat!="ALL")
+    {
+        $form_apptcat=intval($_POST['form_apptcat']);
+    }
+}
+
 $appointments = array();
 $from_date = date("Y-m-d");
 $to_date = date("Y-m-d");
 $datetime = date("Y-m-d H:i:s");
 
 # go get the information and process it
-$appointments = fetch_Patient_Tracker_Events($from_date, $to_date);
+$appointments = fetch_Patient_Tracker_Events($from_date, $to_date, $provider, $facility, $form_apptstatus, $form_apptcat);
 $appointments = sortAppointments( $appointments, 'time' );
 
+//grouping of the count of every status
+$appointments_status = getApptStatus($appointments);
+$lres = sqlStatement("SELECT option_id, title FROM list_options WHERE list_id = ? AND activity=1", array('apptstat'));
+while ( $lrow = sqlFetchArray ( $lres ) ) {
+    $splitTitle = explode(' ', $lrow['title']);
+    $statuses_list[$lrow['option_id']] = $splitTitle[1];
```

Comment:
A better route to take would be:
IF the 2nd character is a space, then do explode, then array_shift, then implode.
ELSE just use the full string.